### PR TITLE
Upgrade autotools, fix compatibility warnings

### PR DESCRIPTION
**Pull request by roystgnr**
This gets us bootstrapping cleanly with the newest autotools, and the newest autotools has nice features like "doesn't fail to configure on M1 and M2 Macs", so let's use it by default now.

### CODE REVIEW

**Comment by lindsayad** [comment]
big fan

**Comment by roystgnr** [comment]
Re-re-bootstrapped to fix the conflicts.

**Review by jwpeterson — APPROVED** [review]
Looks good! Did you test whether, if we run `./bootstrap` without the `--build-autotools` flag, it does the right thing and builds the new versions?

File: m4/ax_tls.m4 [hunk 1]
```diff
@@ -51,14 +51,14 @@ AC_DEFUN([AX_TLS], [
       case $ax_tls_keyword in
         none) ac_cv_tls=none ; break ;;
         *)
-           AC_TRY_COMPILE(
+           AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
```
[the remark this answers is not in the document]
Comment (roystgnr) [reply m4/ax_tls.m4:54]: https://www.gnu.org/software/autoconf-archive/ax_tls.html looks like it's got a 2010 update with this type of change incorporated; I'll try it out.

**Comment by roystgnr** [comment]
Weird.  It's not that libMesh master is automatically inserting a `/usr/include` out of nowhere, it's that we're pointing `$EIGEN_INC` there for some reason, but there's no Eigen there ... but libMesh master still adds /usr/include to our paths anyway, after which point we detect /usr/include/glpk.h and so we get a GLPK we didn't ask for and we enable reduced_basis_ex2 and we get a little more coverage.

I don't actually understand how this PR fixes that.  Trying things out by hand.

**Comment by jwpeterson** [comment]
Maybe the updated autotools fixes a bug that was previously causing `/usr/include` to be added to the `PATH`?

**Comment by roystgnr** [comment]
Maybe ... but investigating this by hand on various cases I'm finding other Eigen detection issues.  I don't want to remove one bug just to add another.

**Comment by roystgnr** [comment]
I was definitely about to be adding another.  Let's see how this theoretically-fixed version works out.

File: m4/ax_tls.m4 (hunk 1 above)
[the remark this answers is not in the document]
Comment (roystgnr) [reply m4/ax_tls.m4]: Done.

File: m4/curl.m4 [hunk 2]
```diff
@@ -46,7 +46,7 @@ AC_DEFUN([CONFIGURE_CURL],
                 [
                   dnl Check for existence of a header file in the specified location
                   curlincFound=no;
-                  AC_CHECK_HEADERS($CURL_INC/curl.h, curlincFound=yes)
+                  AC_CHECK_HEADERS(curl.h, curlincFound=yes, , $CURL_INC)
```
[the remark this answers is not in the document]
Comment (roystgnr) [reply m4/curl.m4]: "Not using a var in the first arg" was indeed a requirement, as was "not completely misinterpreting the documentation on the fourth arg"...

**Comment by roystgnr** [comment]
Okay, those remaining coverage diffs are no longer red flags IMHO.  I'm going to check a few configure logs in CI by hand but if there's nothing crazy there either then I'll merge.

**Pull request merged**